tests/integration/helpers.py

```diff
@@ -236,28 +247,20 @@ def produce_and_check_logs(
         model_full_name=model_full_name,
         endpoint="kafka-client-admin",
     )
-    topic = topic
-    username = relation_data.get("username", None)
-    password = relation_data.get("password", None)
-    servers = relation_data.get("endpoints", "").split(",")
-    security_protocol = "SASL_PLAINTEXT"
-
-    if not (username and password and servers):
-        raise KeyError("missing relation data from app charm")
-
     client = KafkaClient(
-        servers=servers,
-        username=username,
-        password=password,
-        security_protocol=security_protocol,
-    )
-    topic_config = NewTopic(
-        name=topic,
-        num_partitions=5,
-        replication_factor=1,
+        servers=relation_data["endpoints"].split(","),
+        username=relation_data["username"],
+        password=relation_data["password"],
+        security_protocol="SASL_PLAINTEXT",
     )
 
-    client.create_topic(topic=topic_config)
+    if create_topic:
+        topic_config = NewTopic(
+            name=topic,
+            num_partitions=num_partitions,
+            replication_factor=replication_factor,
+        )
+        client.create_topic(topic=topic_config)
     for i in range(15):
```

Comment:
**niptick** can we put the `15` as input, as provide this as an input? I believe it would also make the tests clearer, as we do assert for this and one may wonder why we assert for ("0", "15")? Otherwise it feels 15 is a magic number